constructor/preconda.py

```diff
@@ -87,7 +87,20 @@ def get_final_url(info, url):
             return new_url
     return url
 
+
+def system_info():
+    import constructor, conda, sys
+    return {'constructor': constructor.__version__,
+            'conda': conda.__version__,
+            'platform': sys.platform,
+            'python': sys.version,
+            'python_version': tuple(sys.version_info)}
+
```

Comment:
Should we also dump the nsis version here (for the windows installer)?

Reply:
Not sure if this is the most correct way, but maybe like this?
```
(root) F:\nwani\constructor>%CONDA_PREFIX%\NSIS\makensis.exe /version
v3.01
```
Or maybe we could somehow ask conda to give the exact package version (along with complete build string) of the installed package with the name 'nsis'

Reply:
Not sure how useful this might be, but since we are talking about system info, for macOS, I'd also like to see the value for `sw_vers` and for Windows, the value of: 
```
(root) F:\nwani\constructor>systeminfo | findstr /B /C:"OS Name" /C:"OS Version"
OS Name:                   Microsoft Windows 7 Professional
OS Version:                6.1.7601 Service Pack 1 Build 7601
```

Reply:
In conda 4.4, one way is

```python
pd = PrefixData(target_prefix_path)
prefix_rec = pd.get('nsis')
if prefix_rec:
    print(prefix_rec.version)
```

The `PrefixData` class should have a `query()` method, just like `PackageCacheData` and `SubdirData` (i.e. repodata).  Unfortunately it doesn't currently it looks like.  We could add that though.  And in that case, the preferred way would be

```python
pd = PrefixData(target_prefix_path)
prefix_rec = next(pd.query(MatchSpec('nsis')), None)
if prefix_rec:
    print(prefix_rec.version)
```

In conda 4.3, which will also work for conda 4.4 because of backward compatibility maintained between 4.3 and 4.4 (might not be maintained indefinitely)

```python
from conda.exports import linked_data
prefix_records = linked_data(target_prefix_path).values()
nsis_prefix_rec = next((rec for rec in prefix_records if rec.name == 'nsis'), None)
if nsis_prefix_rec:
    print(nsis_prefix_rec.version)
```